common/src/main/java/org/apache/celeborn/common/write/InFlightRequestTracker.java

```diff
@@ -62,12 +60,14 @@ public void addBatch(int batchId, String hostAndPushPort) {
   public void removeBatch(int batchId, String hostAndPushPort) {
     ConcurrentHashMap<Integer, BatchInfo> batchIdMap =
         inflightBatchesPerAddress.get(hostAndPushPort);
-    BatchInfo info = batchIdMap.remove(batchId);
-    if (info != null && info.channelFuture != null) {
-      info.channelFuture.cancel(true);
-    }
-    if (batchIdMap.size() == 0) {
-      inflightBatchesPerAddress.remove(hostAndPushPort);
+    if (batchIdMap != null) {
```

Comment:
in what case will batchIdMap be null?

Reply:
Trigger on success before call addBatch for push data?
Can resolve this by moving call addBatch method before check locations, WDYT?

Reply:
I think wrappedCallback should only be called after the data is pushed, and that guarantees that the location will not be null.

Reply:
I think in the ```if (loc == null) {``` branch, we should call callback instead of wrappedCallback, WDYT?